### PR TITLE
Add impersonation namespace to system namespaces

### PR DESCRIPTION
**Pull request by cmurphy**
Backport of https://github.com/rancher/rancher/pull/33701

### CODE REVIEW

**Review by prachidamle — APPROVED** [review]

**Comment by prachidamle** [comment]
We dint need this test change of v2.6?

**Comment by cmurphy** [comment]
@prachidamle it passed without it :woman_shrugging:

File: tests/integration/suite/test_system_project.py [hunk 1]
```diff
@@ -11,6 +11,7 @@
                                  "kube-public",
                                  "cattle-global-data",
                                  "cattle-global-nt",
+                                 "cattle-impersonation-system",
```
Comment (prachidamle) [inline tests/integration/suite/test_system_project.py:14]: @cmurphy Maybe can there be a timing issue involved?
- Do we create this "cattle-impersonation-system" namespace during startup always? If yes then this change is needed in 2.6 also?
- Or if it depends on some other usecase to get created, are there chances that some testcase created it? Example this test handles the logging-namespace and removes it from the comparison.

Comment (cmurphy) [reply tests/integration/suite/test_system_project.py:14]: I'm investigating. It looks like this test is run on the local cluster and the namespace shouldn't actually be created there, so we shouldn't need this.

**Review by prachidamle — APPROVED** [review]

**Pull request merged**